### PR TITLE
Expose PersistentList/Mapping docs. Fixes #52.

### LINKED ISSUE #52: Docs: Document PersistentMapping and PersistentList
These aren't referenced from the local docs anywhere. (They're pulled into the ZODB docs on zodb.readthedocs.io for some reason.) Some narrative on why you'd want to use them might be appropriate here.

### PR DESCRIPTION
**Pull request by jamadden**

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -5,6 +5,7 @@
    :maxdepth: 2
 
    api/interfaces
+   api/collections
```
Comment (tseaver) [inline docs/api.rst]: This file seems to be missing from the PR.

File: docs/using.rst [hunk 2]
```diff
@@ -412,33 +417,34 @@ Overriding the attribute protocol
 Subclasses which override the attribute-management methods provided by
 :class:`persistent.Persistent`, but must obey some constraints:
 
-:meth:`__getattribute__``
+
+``__getattribute__``
```
Comment (tseaver) [inline docs/using.rst]: Why are we dropping the `:meth:` role?

File: persistent/list.py [hunk 3]
```diff
@@ -21,6 +21,12 @@
 from persistent._compat import PYTHON2
 
 class PersistentList(UserList, persistent.Persistent):
+    """
```
Comment (tseaver) [inline persistent/list.py]: Summary line follows the triple quote immediately (PEP 257).

File: docs/using.rst (hunk 2 above)
Comment (tseaver) [inline docs/using.rst]: I don't know from that tool.  It looks as though `:method:` is the [currenly-in-vogue spelling](http://www.sphinx-doc.org/en/stable/domains.html#directive-py:method).

Comment (jamadden) [inline docs/using.rst]: It just runs sphinx with the warnings cranked up is all.

But that change is really unrelated, I'll revert.

Comment (jamadden) [inline docs/using.rst]: I suspect probably what it was complaining about was that the :meth: appeared outside of a class body, and [without a leading class name](http://www.sphinx-doc.org/en/stable/domains.html#directive-py:class). So it didn't know what it was trying to describe.

File: docs/conf.py [hunk 4]
```diff
@@ -50,16 +50,16 @@
...
 # The version info for the project you're documenting, acts as replacement for
 # |version| and |release|, also used in various other places throughout the
 # built documents.
 #
 # The short X.Y version.
-version = '3.10'
+version = '4.2'
 # The full version, including alpha/beta/rc tags.
-release = '3.10b1'
+release = '4.2.2'
```
Comment (icemac) [inline docs/conf.py]: This is not DRY. It was forgotten to update for years. The only way to make it DRY seems to be using this `version.txt` approach.

**Pull request merged**